### PR TITLE
reuse catchall implementation

### PR DESCRIPTION
**Pull request by ramaraochavali**
This PR simplifies and reuses the catchall implementation.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by ramaraochavali** [comment]
@rshriram PTAL if this makes sense?

File: pilot/pkg/networking/core/v1alpha3/route/route.go [hunk 1]
```diff
@@ -960,45 +960,6 @@ func getHashPolicy(push *model.PushContext, node *model.Proxy, dst *networking.H
...
-
-const (
-	envoyPath envoyRouteType = iota
-	envoyPrefix
-	envoyRegex
-	envoyCatchAll
-)
-
-// nolint: unparam
-func getEnvoyRouteTypeAndVal(r *route.Route) (envoyRouteType, string) {
```
Comment (ramaraochavali) [inline pilot/pkg/networking/core/v1alpha3/route/route.go]: Currently this is used only for cathall logic - so I removed it and combined with the new function I added in other PR as it has tests and if we need to enhance catchall logic in future if it is at one place. But if this function is intended for any other purpose - I can revert this.

Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/v1alpha3/route/route.go]: I dont think it is a good idea to couple match all route with match all conditions.

Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/v1alpha3/route/route.go]: As the usage, i suggest renaming it to `isCatchAllRoute`

Comment (ramaraochavali) [reply pilot/pkg/networking/core/v1alpha3/route/route.go]: I can rename this - But just curious, why do you think it is not a good idea to combine both?

Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/v1alpha3/route/route.go]: They are different objects and the implemented logic is different.

Comment (ramaraochavali) [reply pilot/pkg/networking/core/v1alpha3/route/route.go]: > They are different objects and the implemented logic is different.

I agree, they are different objects but the implemented logic is same. That was my motivation to combine.

Comment (ramaraochavali) [reply pilot/pkg/networking/core/v1alpha3/route/route.go]: But I can separate them, NBD

**Comment by ramaraochavali** [comment]
@hzxuzhonghu Separated in to two functions. PTAL.

**Pull request merged**